client/src/app/mainwindow.cpp

```diff
@@ -605,8 +611,18 @@ void MainWindow::slotConfigure()
 void MainWindow::slotPrintReport()
 {
     Page *page = currentPage();
-    if (page)
-        page->printReport();
+    if (!page) {
+        return;
+    }
+
+    auto report = page->generateReport();
+
+    KDReports::PreviewDialog preview(report.get(), this);
+    preview.setWindowTitle(i18n("Print Preview"));
+    preview.previewWidget()->setShowPageListWidget(false);
+    preview.previewWidget()->setShowTableSettingsDialog(false);
```

Comment:
(pre-existing) this line is a no-op, fatcrm doesn't use the mode KDReports::Report::SpreadSheet.